dataclients/kubernetes/kube.go

```diff
@@ -467,11 +467,19 @@ func (c *Client) ingressToRoutes(items []*ingressItem) ([]*eskip.Route, error) {
 				if prule.Backend.Traffic > 0 {
 					r, err := c.convertPathRule(i.Metadata.Namespace, i.Metadata.Name, rule.Host, prule, servicesURLs)
 					if err != nil {
-						// tolerate single rule errors
+						// fail when there's a temporary network error so that we keep the last known rules
+						// active until the next sync loop.
+						if err, ok := err.(net.Error); ok {
+							log.Errorf("temporary error while getting service, deferring update: %v", err)
+							return nil, err
```

Comment:
not good. This will delete all the existing routes on update if there is a network error after the initial load was successful.

The checking of the error type should happen after loadAndConvert() was called in LoadAll and LoadUpdate. If it is a network error during LoadAll, then the routes in c.current should be returned. If it is a network error in LoadUpdate, then nil should be returned for the routes and the deleted ids, plus an error.

The reason for the error in the case of LoadUpdate() is that when LoadUpdate() fails, skipper will automatically call LoadAll() and repeat calling it with a delay (debounce without throttling), in order to recover when the source (the ingress API in this case) recovers. (We should document this for the routing.DataClient interface.)